tests/integration/components/denali-progress-test.js

```diff
@@ -0,0 +1,133 @@
+
+    this.set('percent', [0]);
+    assert.dom('.d-progress div').hasAttribute('aria-valuenow', '0', 'DenaliProgress percent should be updated to 0');
+  });
+
+  test('it supports colors', async function (assert) {
+    await render(hbs`
+      <DenaliProgress @percent={{this.percent}} @colors={{this.colors}} />
+    `);
+
+    const redBgStyle = 'width: 30%;background-color: red;';
+    const greenBgStyle = 'width: 30%;background-color: green;';
+    this.set('percent', [30]);
+    assert
+      .dom('.d-progress div')
+      .hasAttribute('aria-valuenow', '30', 'DenaliProgress with default color should be displayed');
+
+    this.set('colors', ['red']);
+    assert
+      .dom('.d-progress div')
+      .hasAttribute('style', redBgStyle, 'DenaliProgress with red color should be displayed');
+
+    this.set('colors', ['green']);
+    assert
+      .dom('.d-progress div')
+      .hasAttribute('style', greenBgStyle, 'DenaliProgress with green color should be displayed');
+  });
+
+  test('it supports isLoading', async function (assert) {
+    await render(hbs`
+      <DenaliProgress @isLoading={{this.isLoading}} @size={{this.size}}/>
+    `);
+
+    assert.dom('div.d-progress').exists('DenaliProgress can render with default `@isLoading` arg set to false ');
+    this.set('isLoading', true);
+    assert
+      .dom('div.d-progress')
+      .hasClass('d-progress__loading', 'DenaliProgress has a loading class when `@isLoading` arg is set to true');
+
+    this.set('size', 'small');
+    assert
+      .dom('div.d-progress')
+      .hasClass('d-progress__sm', 'DenaliProgress has a smaill size when `@size` arg is set to small');
+  });
+
+  test('it supports sizes', async function (assert) {
+    await render(hbs`
+      <DenaliProgress @size={{this.size}}/>
+    `);
+
+    assert.dom('div.d-progress').exists('DenaliProgress can render with default size');
+
+    this.set('size', 'small');
+    assert
+      .dom('div.d-progress')
+      .hasClass('d-progress__sm', 'DenaliProgress has a smaill size when `@size` arg is set to small');
+
+    this.set('size', 'medium');
+    assert
+      .dom('div.d-progress')
+      .hasClass('d-progress__md', 'DenaliProgress has a medium size when `@size` arg is set to medium');
+
+    this.set('size', 'large');
+    assert
+      .dom('div.d-progress')
+      .hasClass('d-progress__lg', 'DenaliProgress has a large size when `@size` arg is set to large');
+  });
+
+  test('it supports multi level progress with colors', async function (assert) {
+    await render(hbs`
+      <DenaliProgress @percent={{this.percent}} @colors={{this.colors}} />
+    `);
+
+    this.set('percent', [30, 40]);
+    this.set('colors', ['orange', 'yellow']);
+
+    const nestedEle = findAll('.d-progress div');
+    console.log(nestedEle, nestedEle[0].style.width, nestedEle[0].style.backgroundColor);
+
+    assert.equal(nestedEle[0].ariaValueNow, 30, 'DenaliProgress with 1st bar with value 30 should be displayed');
```

Comment:
The `assert.dom` calls can take a string or Element as an argument so you could replace this with
```suggestion
    assert.dom(nestedEle[0]).hasAttribute('aria-valuenow', '30', 'DenaliProgress with 1st bar with value 30 should be displayed');
```